astropy/units/core.py

```diff
@@ -886,15 +886,23 @@ def get_err_str(unit):
             f"{unit_str} and {other_str} are not convertible")
 
     def _get_converter(self, other, equivalencies=[]):
-        other = Unit(other)
```

Comment:
Removal of this is helpful to ensure the use in ``quantity_helpers`` remains as fast as possible.